### ISSUE TITLE
Getting 0 as prediction in Classification 

### ISSUE THREAD

**Issue by caprioGirl** [vqdang/hover_net#179]
when training with CoNSeP i am getting 0 at type 1 dice after nearly every 20-23 epochs of stage 2 training. 
![Screenshot from 2021-11-23 15-59-53](https://user-images.githubusercontent.com/26341445/143012765-0bb624b6-6da4-49b0-b00f-ee96f5f91c6a.png)
This behavior also happens with the PanNuke Dataset

**Comment by vqdang** [comment]
Hi, I am sorry for the late reply. The above can happen when the number of instances of type 1 for training is small. I think it should be the Miscellaneous type. You should try to split your training set so that it contains more of that type.

**Comment by caprioGirl** [comment]
yes i figured much, that it's because of the miscellaneous type. I am using whole of CoNSeP training data just like in the paper, so I am confused why it is going 0? am i doing something wrong? did you guys split the training data wrt the misc type to get the results?

**Comment by simongraham** [comment]
Miscellaneous is a very noisy class. Sometimes you will get some unexpected behaviour. Do you have the dice over time during training? You should have this in your tensorboard output.

**Comment by caprioGirl** [comment]
Here is the dic
![Screenshot from 2021-12-08 14-00-47](https://user-images.githubusercontent.com/26341445/145179681-728c8223-6667-4648-a7c7-f587f9527c25.png)
e

**Comment by caprioGirl** [comment]
am i missing something in the classification pipeline?? @simongraham

**Comment by namin0-0** [comment]
@simongraham

**Comment by hjtalent2023** [comment]
Hello! I have met a similarly question with you. like flow:
------valid-np_acc    : 0.92795
------valid-np_dice   : 0.73184
------valid-tp_dice_0 : 0.94518
------valid-tp_dice_1 : 0.02475
------valid-tp_dice_2 : 0.49475
------valid-tp_dice_3 : 0.00000
------valid-tp_dice_4 : 0.61169
------valid-hv_mse    : 0.06671
the dice_3 is always 0. I use the Consep dataset,and set the nr_types are 5.
